shared/src/store/action-reducers/utils/log.ts

```diff
@@ -43,6 +96,469 @@ export function logPatient(
+                ? (target as TransferPoint).externalName
+                : (target as Hospital).name
+        } eingerichtet.`,
+        transferSourceId
+    );
+}
+
+export function logTransferPointConnectionRemoved(
+    state: Mutable<ExerciseState>,
+    transferSourceId: UUID,
+    transferTargetId: UUID,
+    transferTargetType: 'hospital' | 'transferPoint'
+) {
+    if (!logActive(state)) return;
+
+    const source = getElement(state, 'transferPoint', transferSourceId);
+    const target = getElement(state, transferTargetType, transferTargetId);
+
+    logTransferPoint(
+        state,
+        [
+            transferTargetType === 'transferPoint'
+                ? isInSimulatedRegion(target as WithPosition)
+                    ? createSimulatedRegionTag(
+                          state,
+                          currentSimulatedRegionOf(
+                              state,
+                              target as WithPosition
+                          ).id
+                      )
+                    : createTransferPointTag(state, transferTargetId)
+                : createHospitalTag(state, transferTargetId),
+        ],
+        `Der Transfer von ${source.externalName} nach ${
+            transferTargetType === 'transferPoint'
+                ? (target as TransferPoint).externalName
+                : (target as Hospital).name
+        } wurde entfernt.`,
+        transferSourceId
+    );
+}
+
+export function logElementAddedToTransfer(
+    state: Mutable<ExerciseState>,
+    transferSourceId: UUID,
+    transferSourceType: 'alarmGroup' | 'transferPoint',
+    elementId: UUID,
+    elementType: 'personnel' | 'vehicle',
+    transferTargetId: UUID,
+    transferTargetType: 'hospital' | 'transferPoint',
+    duration: number
+) {
+    if (!logActive(state)) return;
+
+    const target = getElement(state, transferTargetType, transferTargetId);
+    const element = getElement(state, elementType, elementId);
+
+    if (transferSourceType === 'alarmGroup') {
+        const alarmGroup = getElement(state, 'alarmGroup', transferSourceId);
+
+        logAlarmGroup(
+            state,
+            [
+                transferTargetType === 'transferPoint'
+                    ? isInSimulatedRegion(target as WithPosition)
+                        ? createSimulatedRegionTag(
+                              state,
+                              currentSimulatedRegionOf(
+                                  state,
+                                  target as WithPosition
+                              ).id
+                          )
+                        : createTransferPointTag(state, transferTargetId)
+                    : createHospitalTag(state, transferTargetId),
+            ],
+            `Ein ${
+                elementType === 'personnel'
+                    ? (element as Personnel).personnelType
+                    : (element as Vehicle).name
+            } wird in ${duration}ms von ${alarmGroup.name} nach ${
```

Comment:
Please format this duration. There is a `formatDuration` in the frontend, which could be extracted to the shared package.

Reply:
d